process-manager.js

```diff
@@ -43,18 +43,14 @@ class ProcessWrapper extends EventEmitter {
 }
 
 class ProcessManager {
-	constructor(options) {
-		if (!options) options = {};
-
+	constructor(execFile, maxProcesses, isChatBased) {
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = '' + execFile;
+		this.maxProcesses = (!isNaN(maxProcesses) || maxProcesses >= 1) ? maxProcesses : 1;
+		this.isChatBased = !!isChatBased;
 
```

Comment:
Ooh, ok, I see what you're talking about now. That would make it less obscure compared to `new ProcessManager(__filename, 4, false)``. I thought you were still meaning something along the lines of using `Object.assign`, but an object that the constructor could only grab the values it needs from could help optimize it and make it easier to understand when it's used elsewhere in the code.